src/main/java/com/diffblue/deeptestutils/Reflector.java

```diff
@@ -366,22 +366,22 @@ private static void makePublic(final CtClass c) {
       throw new DeeptestUtilsRuntimeException(e.getMessage(), e.getCause());
     }
 
-    for (CtMethod m : cl.getDeclaredMethods()) {
-      makePublic(m);
-    }
+    // we consider a class abstract if any method has no body
+    if (isAbstract(cl) || cl.isInterface()) {
```

Comment:
I'm pretty sure this is safe: for a class with a private method, e.g.
```
public class MyClass {
  private int privateMethod() {
    return 12;
  }
}
```
a test that directly calls the private method, e.g.
```
public void myTest throws InvocationTargetException {
    MyClass my = (MyClass) Reflector.getInstance("MyClass");
    int retval = my.privateMethod();
    Assert.assertEquals(retval, 12);
 }
```
would never have compiled. `Reflector.getInstance` only sets attributes of the `CtClass` object representing the given class to public, but it doesn't change anything about the actual `Class` object, and even if it could do that, the Java compiler wouldn't know this at compile time. As far as I can see, the only reason why we're setting things to public here is to allow inheritance.

Reply:
I assumed in the above case we would generate code to call the method using reflection, but that this might cause a run time error if the method itself hasn't been set to public at runt time.

Reply:
We always call private methods using a sequence of calls like
```
  final Class<?> classUnderTest = Reflector.forName("MyClass");
  final Method methodUnderTest = classUnderTest.getDeclaredMethod("privateMethod");
  methodUnderTest.setAccessible(true);
  final int retval = (int) methodUnderTest.invoke(objectUnderTest);
```
The `setAccessible` line gives us access to the private method. This line has always been necessary: again, the body of `Reflector.getInstance` has only ever changed the attributes of the `CtClass` to make them inheritable; it does not modify the `Class` object that we get the `Method` from. I will add a regression test in a new file `ReflectorForNameTest` for calling private methods of objects constructed using reflection as above.